src/Domain/CoordinateMaps/TimeDependent/Shape.cpp

```diff
@@ -335,6 +336,218 @@ tnsr::Ij<tt::remove_cvref_wrap_t<T>, 3, Frame::NoFrame> Shape::jacobian(
   return result;
 }
 
+void Shape::coords_frame_velocity_jacobian(
+    gsl::not_null<std::array<DataVector, 3>*> target_coords,
```

Comment:
To avoid duplicating code and potentially introducing a bug in the future when only one function is updated but another isn't, I think you should factor out the bulk of the `operator()`, frame velocity, and jacobian functions into free functions that take some common arguments (like centered coords, theta_phis, functions of time)